src/config.c

```diff
@@ -417,6 +425,26 @@ int load_config(struct vpn_config *cfg, const char *filename)
 			cfg->user_cert = strdup(val);
 			if (strncmp(cfg->user_cert, "pkcs11:", 7) == 0)
 				cfg->use_engine = 1;
+		} else if (strcmp(key, "dns") == 0) {
+			free(cfg->domain_suffix);
+			cfg->domain_suffix = val;
+			continue;
+		} else if (strcmp(key, "dns") == 0) {
+			Node *newNode = (Node *)malloc(sizeof(Node));
+			newNode->value = val;
+			newNode->next = cfg->dns;
+			cfg->dns = newNode;
+			continue;
+		} else if (strcmp(key, "route") == 0) {			
+    		Node *newNode = (Node *)malloc(sizeof(Node));
+			newNode->value = cidr_to_ip_mask(val);
+			newNode->next = cfg->routes;
+			cfg->routes = newNode;
+			continue;
+		} else if (strcmp(key, "saml-login") == 0) {
+			free(cfg->saml_port);
+			cfg->saml_port = atol(val);
```

Comment:
Are we freeing an int here? And why atol instead of atoi when saml_port is actually an int?